02_calculator/calculator.rb

```diff
@@ -9,7 +9,26 @@ def subtract(first, second)
 end
 
 def sum(array)
-  summ = 0
-  array.each { |x| summ += x }
-  summ
+  array.reduce(0) { |summ, n| summ + n }
+end
+
+def multiply(numbers)
+  numbers.reduce(1) { |prod, n| prod * n }
+end
+
+def power(number, power)
+  number**power
+end
+
+def factorial(number)
+  response = 1
+  if number == 0
+    return response
+  end
+
+  while number != 1
+    response *= number
+    number -= 1
+  end
+  return response
```

Comment:
Return is optional in ruby. Result from last line is returned from the function, without the need to explicitly say `return`